### PR TITLE
Resolve False-Positive in Check 98

### LINKED ISSUE #893: Check 98 - Combine empty catches - false positive - no try catch
minimal example:
```abap
report ztest.
data gv_i type i.

if gv_i is initial.
  write: 'no catch and not empty'.
else.
  write: 'no catch and not empty, too'.
endif.
```

"Empty catches should be combined" on Line 4 (`if gv_i is initial.`)

### PR DESCRIPTION
**Pull request by DennstedtB**
resolves #893

### CODE REVIEW

File: src/checks/zcl_aoc_check_98.clas.abap [hunk 1]
```diff
@@ -40,11 +40,24 @@ CLASS ZCL_AOC_CHECK_98 IMPLEMENTATION.
     DATA ls_statement LIKE LINE OF io_scan->statements.
     DATA ls_struc_index TYPE ty_struc_index_s.
     DATA lt_struc_index TYPE ty_struc_index_tt.
+    DATA lt_deletion_table TYPE ty_struc_index_tt.
 
     lt_struc_index = get_structure_index( it_structures = io_scan->structures ).
 
+    " Delete all non-catch structures
     DELETE lt_struc_index WHERE type <> 'C'.
```
Comment (g-back) [inline src/checks/zcl_aoc_check_98.clas.abap]: could you use the constant for this?  `zcl_aoc_scan=>gc_structure-condition`

File: src/checks/zcl_aoc_check_98.clas.abap [hunk 2]
```diff
@@ -40,11 +40,24 @@ CLASS ZCL_AOC_CHECK_98 IMPLEMENTATION.
...
+    LOOP AT lt_struc_index INTO ls_struc_index.
+      IF ls_struc_index-stmnt_type <> '+'.
```
Comment (g-back) [inline src/checks/zcl_aoc_check_98.clas.abap]: `zcl_aoc_scan=>gc_structure_statement-catch`

**Comment by g-back** [comment]
I had another false positive when using method-calls in the catch blocks.
Could you also please add the following test and see if your solution fixes it?
```abap
    " try-catch with method call
    _code 'TRY.'.
    _code '  CHECK 1 = 2.'.
    _code '  CATCH CX_STATIC_CHECK.'.
    _code '    method_call( 1 ).'.
    _code '  CATCH CX_ROOT.'.
    _code '    another_method_call( 1 ).'.
    _code 'ENDTRY.'.

    ms_result = zcl_aoc_unit_test=>check( mt_code ).

    cl_abap_unit_assert=>assert_initial( act = ms_result-code ).
```

**Comment by DennstedtB** [comment]
Nope it doesn't. I'll close this PR and will have a look at my check again..

**Pull request closed without merging**